File: KCD/HMApplySuppliesCommand.m
//
//  HMApplySuppliesCommand.m
//  KCD
//
//  Created by Hori,Masaki on 2014/09/24.
//  Copyright (c) 2014年 Hori,Masaki. All rights reserved.
//

#import "HMApplySuppliesCommand.h"

#import "HMKCShipObject+Extensions.h"
#import "HMServerDataStore.h"


@implementation HMApplySuppliesCommand
- (void)execute
{
	HMServerDataStore *store = [HMServerDataStore oneTimeEditor];
	
	NSDictionary *data = self.json[@"api_data"];
	NSArray *shipInfos = data[@"api_ship"];
	
	for(NSDictionary *updataInfo in shipInfos) {
		NSError *error = nil;
		NSArray<HMKCShipObject *> *ships = [store objectsWithEntityName:@"Ship"
																  error:&error
														predicateFormat:@"id = %@", @([updataInfo[@"api_id"] integerValue])];
		if(ships.count == 0) {
			if(error) {
				NSLog(@"Error: at %@ : %@", NSStringFromClass([self class]), error);
			}
			continue;
		}
		
		ships[0].bull = updataInfo[@"api_bull"];
		ships[0].fuel = updataInfo[@"api_fuel"];
		NSArray *onslots = updataInfo[@"api_onslot"];
		ships[0].onslot_0 = onslots[0];
		ships[0].onslot_1 = onslots[1];
		ships[0].onslot_2 = onslots[2];
		ships[0].onslot_3 = onslots[3];
		ships[0].onslot_4 = onslots[4];
	}
}
@end

File: KCD/HMApplySuppliesCommand.swift
//
//  HMApplySuppliesCommand.swift
//  KCD
//
//  Created by Hori,Masaki on 2015/01/16.
//  Copyright (c) 2015年 Hori,Masaki. All rights reserved.
//

import Cocoa

class HMApplySuppliesCommand: HMJSONCommand {
	override func execute() {
		if let data = json["api_data"] as? [NSObject : AnyObject] {
			if let shipInfos = data["api_ship"] as? [AnyObject] {
				let store = HMServerDataStore.oneTimeEditor()
				
				for shipInfo in shipInfos {
					if let shipID = shipInfo["api_id"] as? NSNumber {
						var error: NSError? = nil
						let array = store.objectsWithEntityName(
							"Ship",
							sortDescriptors: nil,
							predicate: NSPredicate(format: "id = \(shipID)"),
							error: &error)
						if array.count == 0 {
							if error != nil {
								NSLog("\(__FUNCTION__) error: \(error!)")
							}
							continue
						}
						
						if let ship = array[0] as? HMKCShipObject {
							if let bull = shipInfo["api_bull"] as? NSNumber {
								ship.bull = bull
							}
							if let fuel = shipInfo["api_fuel"] as? NSNumber {
								ship.fuel = fuel
							}
							if let onslot = shipInfo["api_onslot"] as? [NSNumber] {
								ship.onslot_0 = onslot[0]
								ship.onslot_1 = onslot[1]
								ship.onslot_2 = onslot[2]
								ship.onslot_3 = onslot[3]
								ship.onslot_4 = onslot[4]
							}
						}
					}
				}
			}
		}
	}
}
